atcoder/internal_math.hpp

```diff
@@ -15,14 +15,14 @@ constexpr long long safe_mod(long long x, long long m) {
     return x;
 }
 
-// Fast moduler by barrett reduction
+// Fast modular multiplication by barrett reduction
 // Reference: https://en.wikipedia.org/wiki/Barrett_reduction
 // NOTE: reconsider after Ice Lake
 struct barrett {
     unsigned int _m;
     unsigned long long im;
 
-    // @param m `1 <= m`
+    // @param m `1 <= m <= 2^31`
```

Comment:
```suggestion
    // @param m `1 <= m < 2^31`
```

We only need to promise this for the range of `int`.